### DISCUSSION TITLE
How to change port number

### DISCUSSION THREAD

**Discussion by Kitasan-Black-Home** [sqlpage/SQLPage#479] [Q&A] [answered]
Good morning,

I will start by saying I have attempted to both search for previous posts as well as use trial and error to try and get a fresh setup to change to a different port.  the sqlpage.json file did not work using the "listen_on" and neither did creating a .env file and adding it in there.  It is possible since I do not know how to code in json that I am doing it wrong in there, however the site does not (from what I have seen) show an example on where and what ALL to include to accomplish changing the default port number.  I am using Windows if that matters at all.

**Comment by lovasoa** [comment, marked as answer]
Hello and welcome to SQLPage !
  Here is an example configuration file to place in `sqlpage/sqlpage.json` that changes the port number to 9090 (from the default 8080):

```json
{
  "port": 9090
}
```

(we have a [list of configuration options](https://github.com/lovasoa/SQLpage/blob/main/configuration.md#configuring-sqlpage) in the documentation)

If that does not work, then can you please post your exact folder layout, your configuration file, and the logs from SQLPage that you should be seeing in your terminal ?

**Reply by Kitasan-Black-Home** [reply]
Thank you for the quick reply.  I had been trying to place it outside in its own curly braces as well as putting quotation marks around the port number.  Once I tried putting it inside then it finally worked!  Thank you again!!

```
{
  "database_url": "sqlite://./sqlpage/sqlpage.db?mode=rwc",
  "port": 50001
}
```

**Reply by lovasoa** [reply]
You are welcome ! 

What are you building with SQLPage, by the way ?

**Reply by Kitasan-Black-Home** [reply]
I was attempting to build an inventory management system for our warehouse at our datacenter as well as a an alternate way to keep track of our walkthroughs.  I may end up coming back to it once more people create installation walkthroughs on youtube showing fresh installs and connecting mysql or other databases.   I am sure I can poke around and eventually get it set up correctly, however by the time that happens I will have no idea what I changed to make everything work lol.  But I will keep an eye on it and hopefully take another go by the end of the year.

**Reply by lovasoa** [reply]
Hey, I would be happy to walk you through the configuration steps. there shouldn't be much to do, and seeing where new users struggle is very valuable for me to improve the documentation. 

feel free to send me an email at contact@ophir.dev and we can setup a call!
